src/models/StyledComponent.js

```diff
@@ -236,7 +238,27 @@ export default (ComponentStyle: Function) => {
         ? `${escape(options.displayName)}-${options.componentId}`
         : options.componentId || componentId
 
-    const componentStyle = new ComponentStyle(rules, attrs, styledComponentId)
+    let finalAttrs = attrs
+
+    // fold the underlying StyledComponent attrs up (implicit extend)
+    // $FlowFixMe
+    if (targetIsSC && target.attrs) finalAttrs = { ...target.attrs, ...attrs }
```

Comment:
I'm starting to think it'd make sense to abstract `attrs` into a shareable set of utilities or sth... Maybe a future thing.

nit: Btw, I don't think having the `let`s here makes sense since a ternary would also do the trick and would imply that it won't change later in this fn